docs/how-to/contribute-tests.rst

```diff
@@ -0,0 +1,118 @@
+
+    print('Hello')  # Code to test
+    Hello               # CPython output
+    Hello               # Batavia output
+    # Outputs match. Test passes!
+
+This test structure is simple and effective. It's used in almost every test we've written.
+
+Adding Tests
+------------
+
+In many cases, existing tests will not cover everything. Feel free to add your own!
+
+The tests corresponding to Batavia implementations of built-ins are available inside
+``tests/builtins``. The Batavia test infrastructure includes a system to check the compatibility of
+JavaScript implementation of Python with the reference CPython implementation.
+
+These tests all derive from ``TranspileTestCase``, which handles running your code in both interpreters
+and comparing outputs. For an example, look at the ``test_bool.py`` file in ``tests/builtins``. You 
+will see two classes with test cases, ``BoolTests`` and ``BuiltinBoolFunctionTests``. Both derive 
+from ``TranspileTestCase``.
+
+Let's look at some test code that checks if a the Batavia implementation of ``bool`` can handle a
+bool-like class that implements ``__bool__``.
+
+.. code-block:: Python
+
+    def test_bool_like(self):
+        self.assertCodeExecution("""
+            class BoolLike:
+                def __init__(self, val):
+                    self.val = val
+
+                def __bool__(self):
+                    return self.val == 1
+            print(bool(BoolLike(0)))
+            print(bool(BoolLike(1)))
+            """)
+
+The ``assertCodeExecution`` method will run the code provided to it in both implementations. This
+code needs to generate some output so that the output can be compared, hence the need to print the
+values. **Code that is not being printed is not being tested.**
+
+Template
+--------
+
+.. code-block:: python
+
+    def test_<builtin>_<feature/case>(self):
+        # Valid Python code to be tested.
+        code = """
+            print('>>> print(<code>)')
+            print(<code>)
+        """
+        self.assertCodeExecution(code)
+
+This code block provides a printout of the code being run as well as the output of the code,
+which can be very useful for debugging in test cases where a decent amount of code is being run.
+
+Testing for Errors
+------------------
+
+Since we're testing the compiler, we need to ensure that errors for all of the builtins are thrown correctly.
+We also want to ensure that we're not getting the wrong errors in our tests. Simply include a try/except
+block in your test.
+
+.. code-block:: python
+
+    def test_some_error(self):
+        code = """
+            try:
+                code_that_raises_a_ValueError()
+            except ValueError as err:
+                print(err)
+        """
+        self.assertCodeExecution(code)
+
+Remember to catch the specific error you want, and then print the error. **Code that is not being printed
+is not being tested.**
```

Comment:
Also: We need to output something on *success* as well - because we need to differente between "succeeded" and "failed but then failed again when printing the error".